app/controllers/nsm/steps/disbursement_cost_controller.rb

```diff
@@ -29,7 +32,24 @@ def ensure_disbursement
       end
 
       def additional_permitted_params
-        [:apply_vat]
+        [:apply_vat, :add_another]
+      end
+
+      def new_record?
+        disbursement.disbursement_date.nil? || disbursement.miles.nil?
+      end
+
+      def build_flash
+        if new_record?
+          if params.dig(:nsm_steps_disbursement_cost_form, :add_another) == YesNoAnswer::YES.to_s
+            count = current_application.disbursements.count + 1
+            { success: t('.added', count: count, disbursements: t('.disbursement').pluralize(count)) }
```

Comment:
I don't think this needs the "+ 1" - at the point of a user saying they want to add another disbursement, they have not yet completed the additional disbursement. 

An example of this is if you click add another for the first time, it reports back "You have added 2 disbursements" when you've only added one and are currently trying to add the 2nd